src/parsers/literal.ts

```diff
@@ -19,6 +19,13 @@ const parser: Parser = {
             if (reader.string.substring(begin, end) === literal) {
                 reader.cursor = end;
                 if (reader.peek() === " " || !reader.canRead()) {
+                    helper.addActions(
```

Comment:
This being an action is actually quite awkward, because if there is ambiguity, then what should happen? Text can only be highlighted in one colour.

Additionally, what about in the case of failures. Do we include the mangled scopes. What about if parsing one thing failed with some highlighting, but something else suceeded. How do we prioritise the values.
Maybe `helper.merge` could manage this if the `merge` argument fails, but that would make `helper.merge` O(n) of actions.